### PR TITLE
Intgr 256

### PR DESCRIPTION
**Pull request by kmets89**

### CODE REVIEW

File: src/components/SearchModal/SearchResultTabContent.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const resultSwitch = (tab: string, results: SearchResult[]): SearchResult[] => {
+  switch (tab) {
+    case SearchSections.All:
+      return results;
+    case SearchSections.APIReference:
+      return results.filter(result => result.category === "reference");
+    case SearchSections.Guides:
+      return results.filter(result => result.category === "guide");
+    default:
+      return results;
```
Comment (jyalenkatian) [inline src/components/SearchModal/SearchResultTabContent.tsx:24]: Can simplify a bit
```suggestion
    case SearchSections.APIReference:
      return results.filter(result => result.category === "reference");
    case SearchSections.Guides:
      return results.filter(result => result.category === "guide");
    case SearchSections.All:
    default:
      return results;
```

File: src/components/SearchModal/ResultsList.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+};
+
+const buildLink = (path: string, title: string, searchResultID: number, matchTerms: MatchInfo): JSX.Element => {
+    let headerId: string | null = findHeaderId(searchResultID, matchTerms);
+    return headerId === null ?
+        <Link to={path}>{highlightMatch(title, matchTerms)}</Link> :
+        <HashLink to={`${path}#${headerId}`}>{highlightMatch(title, matchTerms)}</HashLink>;
+};
+
+export default function ResultsList(props: { results: SearchResult[] }): JSX.Element {
```
Comment (jyalenkatian) [inline src/components/SearchModal/ResultsList.tsx:29]: Should the props be made into an interface?

File: src/components/SearchModal/utils/searchUtils.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        if ((excerpt.length - excerpt.indexOf(matchTerm) - matchTerm.length) !== numCharsAfter) {
+            excerpt += "...";
+        }
+        return excerpt;
+    }
+    //if there's no match in the page content, return the first portion of the page
+    return contentString.substring(0, EXCERPT_CHAR_LENGTH) + "...";
+};
+
+export const findHeaderId = (searchId: number, matchTerms: MatchInfo,): string | null => {
```
Comment (jyalenkatian) [inline src/components/SearchModal/utils/searchUtils.tsx:60]: Trailing comma
```suggestion
export const findHeaderId = (searchId: number, matchTerms: MatchInfo): string | null => {
```

**Review by jyalenkatian — CHANGES_REQUESTED** [review]
@kmets89 Comments have been added to the PR.

File: src/components/SearchModal/SearchModal.tsx [hunk 4]
```diff
@@ -1,42 +1,81 @@
...
+  searchOptions: {
+    fuzzy: 0.2,
+    prefix: true,
+    boost: {
+      "title": 3,
+      "headers": 2
+    }
+  }
+});
+miniSearch.addAll(searchIndex); //TODO: should we do this asynchronously? how do we handle the UI until it's ready
```
[the remark this answers is not in the document]
Comment (kmets89) [reply src/components/SearchModal/SearchModal.tsx:36]: let's revisit this once all search stuff is done

File: src/components/apiReference/Method.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+export default function Method(): JSX.Element {
+    const { methodId = "" } = useParams();
+    const storedMethodData: MethodData = JSON.parse(sessionStorage[methodId]);
+    const parameters: MethodParameter[] = storedMethodData.parameters;
+    const returnValueDescriptions: string = storedMethodData.returns;
+    const codeSample: string = storedMethodData.example;
+
+    const introductionParagraph: ReactNode = <div className="paragraph">{RenderStringWithUrl(methodId, JSON.parse(sessionStorage[methodId]).description)}</div>;
+
+    const parameterParagaphs: ReactNode = (
```
Comment (jyalenkatian) [inline src/components/apiReference/Method.tsx]: Typo that we missed in the previous branch

```suggestion
    const parameterParagraphs: ReactNode = (
```

File: src/components/apiReference/Method.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    const pageSections: TableOfContentsItem[] = [
+        {
+            elementId: "introduction",
+            summary: "Introduction",
+            details: introductionParagraph
+        },
+        {
+            elementId: "parameters",
+            summary: "Parameters",
+            details: parameterParagaphs
```
Comment (jyalenkatian) [inline src/components/apiReference/Method.tsx]: Typo that we missed in the previous branch

```suggestion
            details: parameterParagraphs
```

File: src/components/apiReference/Object.tsx [hunk 7]
```diff
@@ -8,57 +8,51 @@ import { TableOfContentsItem } from "../TableOfContents/TableOfContents";
...
     const properties: ObjectProperty[] = storedObjectData.properties;
 
-    const introductionParagraph: ReactNode = (
-        <div className="paragraph">
-            {RenderStringWithUrl(objectDescription)}
-        </div>
-    );
+    const introductionParagraph: ReactNode = <div className="paragraph">{RenderStringWithUrl(objectId, objectDescription)}</div>;
 
     const propertyParagaphs: ReactNode = (
```
Comment (jyalenkatian) [inline src/components/apiReference/Object.tsx]: Typo that we missed in the previous branch
```suggestion
    const propertyParagraphs: ReactNode = (
```

File: src/components/apiReference/Object.tsx [hunk 8]
```diff
@@ -8,57 +8,51 @@ import { TableOfContentsItem } from "../TableOfContents/TableOfContents";
...
+            summary: "Introduction",
+            details: introductionParagraph
+        },
         {
-            "elementId": "properties",
-            "summary": "Properties",
-            "details": propertyParagaphs
+            elementId: "properties",
+            summary: "Properties",
+            details: propertyParagaphs
```
Comment (jyalenkatian) [inline src/components/apiReference/Object.tsx]: Typo that we missed in the previous branch

```suggestion
            details: propertyParagraphs
```

**Review by jyalenkatian — CHANGES_REQUESTED** [review]

**Review by jyalenkatian — APPROVED** [review]

**Pull request merged**